lib/jalaali/calendar.ex

```diff
@@ -2,56 +2,48 @@ defmodule Jalaali.Calendar do
   @moduledoc """
   A calendar implementation based on jalaali calendar system
   """
-
-  @behaviour Calendar
-
-  @type year :: -61..3178
-  @type month :: 1..12
-  @type day :: 1..31
-  @type day_of_week :: 1..7
-  @type hour :: 0..23
-  @type minute :: 0..59
-  @type second :: 0..60
-  @type microsecond :: Integer.t()
-
   @seconds_per_minute 60
   @seconds_per_hour 60 * 60
   # Note that this does NOT handle leap seconds.
   @seconds_per_day 24 * 60 * 60
   @microseconds_per_second 1_000_000
-
   @months_in_year 12
-
   # The Jalaali epoch starts, in this implementation,
   # with era 1 on 0001-01-01 which is 227261 days later.
   @jalaali_epoch 227_261
 
+  @behaviour Jalaali.Helper
+
   @impl true
-  @spec days_in_month(Calendar.year(), Calendar.month()) :: Calendar.day()
   def days_in_month(year, month), do: Jalaali.jalaali_month_length(year, month)
 
   @impl true
-  @spec leap_year?(Calendar.year()) :: boolean
   def leap_year?(year), do: Jalaali.is_leap_jalaali_year(year)
 
-  @impl true
-  @spec day_of_week(year, month, day) :: day_of_week
   @doc """
   Returns day of week on a spesific set of year, month and day
   """
-  def day_of_week(year, month, day) do
-    {:ok, date} = Date.new(year, month, day, __MODULE__)
-    iso_date = Date.convert!(date, Calendar.ISO)
-    Calendar.ISO.day_of_week(iso_date.year, iso_date.month, iso_date.day)
+  @impl true
+  if Code.ensure_loaded?(Calendar.ISO) && function_exported?(Calendar.ISO, :day_of_week, 4) do
```

Comment:
We don't need to check for version. We are going to rollout a new major version which will be supported from elixir 1.12+ (or maybe 1.11)

Reply:
If we have users which are using elixir 1.6 and still want to have new version of this lib, is better I think, and we do not need to consider version of elixir because this code just check the function we need is loaded or not